src/module-elasticsuite-catalog/Plugin/Ui/Category/Form/DataProviderPlugin.php

```diff
@@ -116,6 +137,66 @@ private function getFilterableAttributeList($currentCategory)
         return $attributesList;
     }
 
+    /**
+     * Retrieve default store view id.
+     *
+     * @return int
+     */
+    private function getDefaultStoreId()
+    {
+        $store = $this->storeManager->getDefaultStoreView();
+
+        if (null === $store) {
+            // Occurs when current user does not have access to default website (due to AdminGWS ACLS on Magento EE).
+            $store = !empty($this->storeManager->getWebsites()) ? current($this->storeManager->getWebsites())->getDefaultStore() : null;
+        }
+
+        return $store ? $store->getId() : 0;
+    }
+
+    /**
+     * Get store id for the current category.
+     *
+     * @param CategoryInterface $category Category.
+     *
+     * @return int
+     */
+    private function getStoreId(CategoryInterface $category)
+    {
+        $storeId = $category->getStoreId();
+
+        if ($storeId === 0) {
+            $defaultStoreId   = $this->getDefaultStoreId();
+            $categoryStoreIds = array_filter($category->getStoreIds());
+            $storeId          = current($categoryStoreIds);
+            if (in_array($defaultStoreId, $categoryStoreIds)) {
+                $storeId = $defaultStoreId;
+            }
+        }
+
+        return $storeId;
+    }
+
+    /**
+     * Return the query filter to be to applied for the category.
+     *
+     * @param CategoryInterface $category Category
+     *
+     * @return QueryInterface
+     */
+    private function getQueryFilter(CategoryInterface $category)
+    {
+        $query = null;
+
+        $category->setIsActive(true);
+
+        if ($category->getIsVirtualCategory() || $category->getId()) {
+            $query = $category->getVirtualRule()->getCategorySearchQuery($category);
```

Comment:
This will fail if VirtualCategory module is disabled. Can you provide a fallback to a legacy category query in this case ? Based on category.category_id field.

Reply:
I could, but the list of attributes would be inherently wrong in this case.
So far I haven't found any code in VirtualCategory to delegate the query building to.
I need to identify more properly the way things are done there because it's not that obvious.

Reply:
Something like : https://github.com/Smile-SA/elasticsuite/blob/master/src/module-elasticsuite-catalog-optimizer/Model/Optimizer/Preview/CategoryQuery.php#L72